### DISCUSSION TITLE
Use a custom kubeconfig file location

### DISCUSSION THREAD

**Discussion by ogarcia** [kube-rs/kube#1474] [Q&A] [answered]
Let's see if you can give me a hand because I'm a bit lost. As I see when you initialize the client with `try_default()` it is going to look for the _kubeconfig_ file either where the environment variable `${KUBECONFIG}` is defined or to the default location `~/.kube/config`. Is there any way I can tell it where that file is? I have been checking the documentation and I can't find anything about it.

Update: Right now I'm using the method of exporting the environment variable myself with `std::env::set_var("KUBECONFIG", "myfile")`, but I was wondering if there was a more elegant way to do this.

**Comment by clux** [comment]
The local kubeconfig might come from multiple files depending on what the evar is set to because kubeconfigs merge together in a rather strange but defined way.

Is there a reason you want to know the file? This particular file merge behaviour only applies to local development.

**Reply by ogarcia** [reply]
It is because I am making a mini-program in which it can be specified by means of a configuration file where the kubeconfig is located.

The reason for doing this is because in my work they have the habit of handling different kubeconfig files depending on the context instead of one where everything is merged and you change the context. :shrug:

**Reply by ogarcia** [reply]
Well, with the option to define the environment variable myself it works for me, but I was wondering if there was another more direct way.

**Reply by clux** [reply, marked as answer]
There is [`Kubeconfig::read_from`](https://docs.rs/kube/latest/kube/config/struct.Kubeconfig.html#method.read_from) provided you enumerate the available configs yourself.
...which you can hook up to [`Config::from_custom_kubeconfig`](https://docs.rs/kube/latest/kube/struct.Config.html#method.from_custom_kubeconfig)

**Reply by ogarcia** [reply]
Thank you very much @clux !!
